orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py

```diff
@@ -88,9 +88,19 @@ def _export_pt_1_10(g, n, *args, **kwargs):
         training_mode = None
         runtime_pytorch_version = version.parse(torch.__version__.split("+")[0])
         if runtime_pytorch_version >= version.parse("1.12"):
+            # FIXME: using privated modules
             from torch.onnx import _globals
 
-            training_mode = _globals.GLOBALS.training_mode
+            # before https://github.com/pytorch/pytorch/commit/c8b9b6266b505328e503b12f6a42fd88c56374f9, training_mode is still a bool type
+            if isinstance(_globals.GLOBALS.training_mode, bool):
+                training_mode = _globals.GLOBALS.training_mode
+            else:
+                if _globals.GLOBALS.training_mode not in [
+                    torch.onnx.TrainingMode.EVAL,
+                    torch.onnx.TrainingMode.TRAINING,
+                ]:
+                    raise Exception(f"Unexpected training mode {_globals.GLOBALS.training_mode}")
+                training_mode = 1 if _globals.GLOBALS.training_mode == torch.onnx.TrainingMode.TRAINING else 0
```

Comment:
nit: I think this might be better, then `training_mode` will be bool in both branches. 

"training_mode = _globals.GLOBALS.training_mode == torch.onnx.TrainingMode.TRAINING"